source/ranges.tex

```diff
@@ -33,154 +33,158 @@
-  class subrange;
+  class subrange;                                                                   // freestanding
 
   template<class I, class S, subrange_kind K>
-    inline constexpr bool enable_borrowed_range<subrange<I, S, K>> = true;
+    inline constexpr bool enable_borrowed_range<subrange<I, S, K>> = true;          // freestanding
 
   // \ref{range.dangling}, dangling iterator handling
-  struct dangling;
+  struct dangling;                                                                  // freestanding
 
   template<@\libconcept{range}@ R>
-    using borrowed_iterator_t = @\seebelow@;
+    using borrowed_iterator_t = @\seebelow@;                                          // freestanding
 
   template<@\libconcept{range}@ R>
-    using borrowed_subrange_t = @\seebelow@;
+    using borrowed_subrange_t = @\seebelow@;                                          // freestanding
 
   // \ref{range.utility.conv}, range conversions
   template<class C, @\libconcept{input_range}@ R, class... Args> requires (!@\libconcept{view}@<C>)
-    constexpr C to(R&& r, Args&&... args);
+    constexpr C to(R&& r, Args&&... args);                                          // freestanding
   template<template<class...> class C, @\libconcept{input_range}@ R, class... Args>
-    constexpr auto to(R&& r, Args&&... args);
+    constexpr auto to(R&& r, Args&&... args);                                       // freestanding
   template<class C, class... Args> requires (!@\libconcept{view}@<C>)
-    constexpr auto to(Args&&... args);
+    constexpr auto to(Args&&... args);                                              // freestanding
   template<template<class...> class C, class... Args>
-    constexpr auto to(Args&&... args);
+    constexpr auto to(Args&&... args);                                              // freestanding
 
   // \ref{range.empty}, empty view
   template<class T>
     requires is_object_v<T>
-  class empty_view;
+  class empty_view;                                                                 // freestanding
 
   template<class T>
-    inline constexpr bool enable_borrowed_range<empty_view<T>> = true;
+    inline constexpr bool enable_borrowed_range<empty_view<T>> = true;              // freestanding
 
   namespace views {
     template<class T>
-      inline constexpr empty_view<T> @\libmember{empty}{views}@{};
+      inline constexpr empty_view<T> @\libmember{empty}{views}@{};                                       // freestanding
   }
 
   // \ref{range.single}, single view
   template<@\libconcept{copy_constructible}@ T>
     requires is_object_v<T>
-  class single_view;
+  class single_view;                                                                // freestanding
 
-  namespace views { inline constexpr @\unspec@ single = @\unspec@; }
+  namespace views {
+    inline constexpr @\unspec@ single = @\unspec@;                              // freestanding
+  }
```

Comment:
I do not think it is generally necessary to single out an entity on a line to mark it as freestanding. [freestanding.entity]p3 says:

> In a header synopsis, entities followed with a comment that includes freestanding are freestanding entities.

In this case, this implies that both `views` and `single` are freestanding entities. The former is already implied by [freestanding.entity]p6, such that singling out `single` is unnecessary:

>  Enclosing namespaces of freestanding entities are freestanding entities.

Reply:
I read this as saying that, since single is freestanding, so is views.  As the editor, my concern here is whether I applied the paper correctly.

Reply:
It is indeed correctly applied.

But moving a declaration to its own line, just to separate it from its namespace, to then tag it as freestanding, is not necessary. That's because marking the declaration as freestanding also makes its enclosing namespace freestanding.
So marking a line with a declaration is enclosed in a namespace on the same line as freestanding, like `namespace x { int y; } // freestanding` has the same effects as
```C++
namespace x {
  int y; // freestanding
}
```

Reply:
"marking the declaration as freestanding also makes its enclosing namespace freestanding" yes, but I don't see where we get the reverse, that is that marking the enclosing namespace freestanding makes its member declarations freestanding. Granted, I haven't studied the paper closely...

Reply:
> that is that marking the enclosing namespace freestanding makes its member declarations freestanding.

That doesn't happen.
What I'm saying is that, since `namespace x { int y; } // freestanding` and
```C++
namespace x {
  int y; // freestanding
}
```
both make `x` and `y` freestanding entities, there's no need to isolate `y` to its own line.
In other words, it should have been enough to append `// freestanding` to
```LaTeX
  namespace views { inline constexpr @\unspec@ single = @\unspec@; }
```
Rather than what's actually done:
```LaTeX
  namespace views {
    inline constexpr @\unspec@ single = @\unspec@;                              // freestanding
  }
```

Reply:
Sorry, I'm just not getting it.  What rule says that if the namespace is freestanding, that its members are?  i.e. what says that 
`namespace x { int y; } // freestanding
`implies that y is freestanding?

Reply:
In any case, this shouldn't stop @tkoeppe from applying these edits as is. As you stated, "It is indeed correctly applied."

Note that I am sympathetic to your concerns, as all these extra "freestanding"s do add quite a bit of noise.

Reply:
done

Reply:
> Are we sure this is what was intended?

I wouldn't know, though I don't see the harm, either.